Maps numbers to names in TSGroupModel

mdouglass opened the pull request:
# Use an injected ContactsManager to look up proper user names for phone numbers in [TSGroupModel getInfoStringAboutUpdateTo].

Fixes WhisperSystems/Signal-iOS#757

// FREEBIE

mdouglass committed:
Maps numbers to names in TSGroupModel

Use an injected ContactsManager to look up proper user names for phone numbers in [TSGroupModel getInfoStringAboutUpdateTo].
Fixes WhisperSystems/Signal-iOS #757

// FREEBIE

mdouglass commented:
This is my first commit to this project, so feel free to beat up on it.  I think I've properly following the contribution guidelines and I've submitted a CLA. Let me know if I missing anything else.

michaelkirk commented on src/Messages/TSGroupModel.h:
```diff
@@ -19,7 +20,7 @@
 - (BOOL)isEqual:(id)other;
 - (BOOL)isEqualToGroupModel:(TSGroupModel *)model;
-- (NSString *)getInfoStringAboutUpdateTo:(TSGroupModel *)model;
+- (NSString *)getInfoStringAboutUpdateTo:(TSGroupModel *)model withContactsManager:(id<ContactsManagerProtocol>)contactsManager;
```
style nit, name the parameter without the preposition, just `contactsManager:(id<ContactsManagerProtocol>)contactsManager`

michaelkirk reviewed (commented):
Thanks for this.

This works when someone else updates the group.

Beyond your improvement here, it would be good to address what happens when _I'm_ the one updating the group. As it is, I see just "Group Updated" in that case.

See: 
`- (void)updateGroupModelTo:(TSGroupModel *)newGroupModel` in MessagesViewController.m for a similar method.

Also, if you're into using cool functions like `map` and injecting dependencies, there's plenty more work to be done in Signal iOS. =)

mdouglass commented:
I'll update the parameter name.  
Should I make the other change in this PR or open another one for that issue?
I'm definitely planning to do more work on Signal-iOS.  Wanted to pick something small to get a foot in the door :)

michaelkirk commented:
Getting group updates that I make to show these details will likely require changes to Signal-iOS, so it will necessitate a PR over there.

I was thinking it made sense to wait on merging this one so you could make sure any shared code works in both cases.

mdouglass commented:
That makes sense, I'll take a look.

michaelkirk commented:
Here's where we're currently handling it
https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/view%20controllers/MessagesViewController.m#L2011

Seemingly as a result of:
https://github.com/WhisperSystems/Signal-iOS/blob/master/Signal/src/view%20controllers/MessagesViewController.m#L2041

mdouglass committed:
Merge branch 'master' into fix-signal-ios-issue-757

mdouglass committed:
Style renaming

mdouglass committed:
Supports custom messages for TSOutgoingMessage

This will enable Signal-iOS to push detailed information about TSGroupModel updates through SignalServiceKit and into the TSInfoMessage created by TSMessagesManager+sendMessages

mdouglass commented:
Ack, something went wrong with that merge before I committed it. I'll have to redo it.

michaelkirk commented:
Feel free to force push your own branches in the future.

mdouglass commented:
Good to know.  Never quite sure when that's ok with git.  Too much time spent in p4.
